cluster/src/main/java/org/apache/iotdb/cluster/server/member/RaftMember.java

```diff
@@ -545,424 +550,487 @@ public Node getThisNode() {
-    if (readOnly) {
-      return StatusUtils.NODE_READ_ONLY;
-    }
+      /**
+       <<<<<<< HEAD
+       * Only the group leader can call this method. Will commit the log locally and send it to
+       * followers
+       *
+       =======
+       * Create a log for "plan" and append it locally and to all followers.
+       * Only the group leader can call this method.
+       * Will commit the log locally and send it to followers
+       >>>>>>> 378da6117d05a7e38d05fb46cb20b3023434c08c
+       * @param plan
+       * @return OK if over half of the followers accept the log or null if the leadership is lost
+       * during the appending
+       */
+      TSStatus processPlanLocally (PhysicalPlan plan){
+        logger.debug("{}: Processing plan {}", name, plan);
+        if (readOnly) {
+          return StatusUtils.NODE_READ_ONLY;
+        }
 
-    PhysicalPlanLog log = new PhysicalPlanLog();
-    // assign term and index to the new log and append it
-    synchronized (logManager) {
-      log.setCurrLogTerm(getTerm().get());
-      log.setPreviousLogIndex(logManager.getLastLogIndex());
-      log.setPreviousLogTerm(logManager.getLastLogTerm());
-      log.setCurrLogIndex(logManager.getLastLogIndex() + 1);
+        PhysicalPlanLog log = new PhysicalPlanLog();
+        // assign term and index to the new log and append it
+        synchronized (logManager) {
+          log.setCurrLogTerm(getTerm().get());
+          log.setPreviousLogIndex(logManager.getLastLogIndex());
+          log.setPreviousLogTerm(logManager.getLastLogTerm());
+          log.setCurrLogIndex(logManager.getLastLogIndex() + 1);
 
-      log.setPlan(plan);
-      logManager.appendLog(log);
-    }
+          log.setPlan(plan);
+          logManager.appendLog(log);
+        }
 
-    if (appendLogInGroup(log)) {
-      return StatusUtils.OK;
-    }
-    return null;
-  }
+        if (appendLogInGroup(log)) {
+          return StatusUtils.OK;
+        }
+        return null;
+      }
 
-  /**
-   * if the node is not a leader, will send it to the leader. Otherwise do it locally (whether to
-   * send it to followers depends on the implementation of executeNonQuery()).
-   *
-   * @param request
-   * @param resultHandler
-   */
-  public void executeNonQueryPlan(ExecutNonQueryReq request,
-      AsyncMethodCallback<TSStatus> resultHandler) {
-    if (character != NodeCharacter.LEADER) {
-      // forward the plan to the leader
-      AsyncClient client = connectNode(leader);
-      if (client != null) {
+      /**
+       <<<<<<< HEAD
+       * if the node is not a leader, will send it to the leader. Otherwise do it locally (whether to
+       * send it to followers depends on the implementation of executeNonQuery()).
+       *
+       =======
+       * Append a log to all followers in the group until half of them accept the log or the
+       * leadership is lost.
+       * @param log
+       * @return true if the log is accepted by the quorum of the group, false otherwise
```

Comment:
Conflict marks.

Reply:
Thanks for your reminding. I will remove them.